src/main/java/io/molnarsandor/trelloclone/user/UserServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package io.molnarsandor.trelloclone.user;
+
+import io.molnarsandor.trelloclone.user.exceptions.ActivationKeyNotFoundException;
+import io.molnarsandor.trelloclone.user.exceptions.UsernameAlreadyExistsException;
+import io.molnarsandor.trelloclone.user.model.UserActivationDTO;
+import io.molnarsandor.trelloclone.user.model.UserEntity;
+import io.molnarsandor.trelloclone.user.model.UserRegisterDTO;
+import io.molnarsandor.trelloclone.user.model.UserRegistrationResponseDTO;
+import io.molnarsandor.trelloclone.util.EmailService;
+import io.molnarsandor.trelloclone.util.ModelConverter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.BindingResult;
+
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class UserServiceImpl implements UserService, UserDetailsService {
+
+    private static final String USER_NOT_FOUND = "User not found";
+
+    private final UserRepository userRepository;
+
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    private final EmailService emailService;
+
+    private final ModelConverter modelConverter;
+
+    private final UserValidator userValidator;
+
+    // == PUBLIC METHODS ==
+    @Transactional
```

Comment:
Ez a `@Transactional` szerintem nem kell ide, egyrészt a metódus read-only, másrészt megnézném hogy a Spring Boot automatikusan "rárakja"-e ezeket vagy sem.

Reply:
Nem feltétlenül probléma ha rajta van viszont akkor a `readOnly` attribútumát fixen állítsd be `true`-ra, hogy biztosan ne lehessen módosítani az entitásokat a metódusban.